engine/src/test/java/org/hibernate/search/engine/backend/orchestration/spi/BatchingExecutorTest.java

```diff
@@ -63,9 +66,11 @@ public class BatchingExecutorTest {
 
 	@Parameterized.Parameters(name = "operation submitter = {0}")
 	public static Object[][] params() {
-		return Arrays.stream( OperationSubmitter.values() )
-				.map( value -> new Object[] { value } )
-				.toArray( Object[][]::new );
+		return new Object[][] {
+				{ "BLOCKING", OperationSubmitter.blocking() },
+				{ "REJECTED_EXECUTION_EXCEPTION", OperationSubmitter.rejecting() },
+				{ "OFFLOADING", OperationSubmitter.offloading( r -> Executors.newFixedThreadPool( 1 ).submit( r ) ) }
```

Comment:
Creating a thread pool and not shutting it down leaks resources. Not critical, I know, but still, this would be more correct:

```suggestion
				{ "OFFLOADING", OperationSubmitter.offloading( CompletableFuture::runAsync ) }
```